File: fix_dashboard_locks.js
const fs = require('fs');
const path = require('path');

const filePath = path.resolve('c:/Users/Lenovo/Desktop/meridone ai life manager/frontend/src/pages/Dashboard.jsx');

let content = fs.readFileSync(filePath, 'utf8');

// Use regex to find the blocks regardless of exact spacing
// Block 1 & 2: Lock overlay check
// Looking for: {stats.user?.plan !== 'Pro' && ( ... Unlock with Pro ... )}
const overlayRegex = /\{stats\.user\?\.plan !== 'Pro' && \(\s*<div className="absolute inset-0 bg-white\/60 backdrop-blur-\[2px\] z-10 flex flex-col items-center justify-center rounded-xl">\s*<div className="flex items-center gap-2 px-4 py-2 bg-gray-900 text-white rounded-full text-xs font-bold shadow-lg">\s*<Lock size=\{12\} \/>\s*Unlock with Pro\s*<\/div>\s*<\/div>\s*\)\}/g;

const overlayReplacement = `{stats.user?.plan !== 'Pro' && localUser?.plan !== 'Pro' && (
                                    <div className="absolute inset-0 bg-white/60 backdrop-blur-[2px] z-10 flex flex-col items-center justify-center rounded-xl">
                                        <div className="flex items-center gap-2 px-4 py-2 bg-gray-900 text-white rounded-full text-xs font-bold shadow-lg">
                                            <Lock size={12} />
                                            Unlock with Pro
                                        </div>
                                    </div>
                                )}`;

// Block 3 & 4: Opacity check
// Looking for: <div className={`space-y-2 ${stats.user?.plan !== 'Pro' ? 'opacity-50 pointer-events-none' : ''}`}>
const opacityRegex = /<div className=\{`space-y-2 \$\{stats\.user\?\.plan !== 'Pro' \? 'opacity-50 pointer-events-none' : ''\}`\}>/g;

const opacityReplacement = `<div className={\`space-y-2 \$\{stats.user?.plan !== 'Pro' && localUser?.plan !== 'Pro' ? 'opacity-50 pointer-events-none' : ''\}\`}>`;

let newContent = content.replace(overlayRegex, overlayReplacement);
newContent = newContent.replace(opacityRegex, opacityReplacement);

if (newContent !== content) {
    fs.writeFileSync(filePath, newContent, 'utf8');
    console.log("Successfully updated Dashboard.jsx");
} else {
    console.log("No matches found. Please check regex accuracy.");
    process.exit(1);
}

File: fix_dashboard_locks.py
import sys

file_path = r'c:\Users\Lenovo\Desktop\meridone ai life manager\frontend\src\pages\Dashboard.jsx'

with open(file_path, 'r', encoding='utf-8') as f:
    content = f.read()

# Replacement 1: Monthly Expenses Lock
target1 = """                            {/* Advanced Insights */}
                            <div className="relative mt-4 pt-4 border-t border-gray-100">
                                {stats.user?.plan !== 'Pro' && (
                                    <div className="absolute inset-0 bg-white/60 backdrop-blur-[2px] z-10 flex flex-col items-center justify-center rounded-xl">
                                        <div className="flex items-center gap-2 px-4 py-2 bg-gray-900 text-white rounded-full text-xs font-bold shadow-lg">
                                            <Lock size={12} />
                                            Unlock with Pro
                                        </div>
                                    </div>
                                )}
                                <div className={`space-y-2 ${stats.user?.plan !== 'Pro' ? 'opacity-50 pointer-events-none' : ''}`}>"""

replacement1 = """                            {/* Advanced Insights */}
                            <div className="relative mt-4 pt-4 border-t border-gray-100">
                                {stats.user?.plan !== 'Pro' && localUser?.plan !== 'Pro' && (
                                    <div className="absolute inset-0 bg-white/60 backdrop-blur-[2px] z-10 flex flex-col items-center justify-center rounded-xl">
                                        <div className="flex items-center gap-2 px-4 py-2 bg-gray-900 text-white rounded-full text-xs font-bold shadow-lg">
                                            <Lock size={12} />
                                            Unlock with Pro
                                        </div>
                                    </div>
                                )}
                                <div className={`space-y-2 ${stats.user?.plan !== 'Pro' && localUser?.plan !== 'Pro' ? 'opacity-50 pointer-events-none' : ''}`}>"""

# Replacement 2: Revenue Tracked Lock (similar block)
# Note: Since the targets are identical, simple replace will work if we do it once or use a different approach.
# However, content.replace(target1, replacement1) will replace ALL occurrences.

new_content = content.replace(target1, replacement1)

if new_content == content:
    print("No changes made. Target not found.")
    sys.exit(1)

with open(file_path, 'w', encoding='utf-8') as f:
    f.write(new_content)

print("Successfully updated Dashboard.jsx")
